### DISCUSSION TITLE
HC-05 + Pi Pico = REPL over Bluetooth????

### DISCUSSION THREAD

**Discussion by petermoscickidev** [micropython/micropython#10283] [RP2040 / Pico]
Hi All,

just out of curiosity - based on my knowledge Micropython for Pi Pico W supports REPL on UART. (correct me if I'm wrong)

Is it possible to connect HC-05 to RX/TX of Pi PIco (UART0) and communicate with Pi Pico from a connected (via Bluetooth, HC-05) Windows PC?? Is it possible to use REPL this way??

Thanks!
Peter

**Comment by jimmo** [comment]
@mostekkk 

> based on my knowledge Micropython for Pi Pico W supports REPL on UART. (correct me if I'm wrong)

That is correct.

> Is it possible to connect HC-05 to RX/TX of Pi PIco (UART0) and communicate with Pi Pico from a connected (via Bluetooth, HC-05) Windows PC?? Is it possible to use REPL this way??

Yes this should be possible.

**Reply by petermoscickidev** [reply]
Thanks for your reply. I noticed, that HC-05 runs by default with 9600. What is the default REPL on UART speed? I assume that it is 115200 or more? Can I customize REPL UART speed?

**Reply by jimmo** [reply]
I'm fairly sure that if you re-init the default UART with a new baud rate, that should work

```
u = machine.UART(0, baudrate=9600)
```

Otherwise you can re-build the firmware with the `PICO_DEFAULT_UART_BAUD_RATE` changed.

(Note that you first need to enable stdio on the default UART using `MICROPY_HW_ENABLE_UART_REPL`).

**Comment by freemansoft** [comment]
https://github.com/micropython/micropython/discussions/10534
